synth.py

```diff
@@ -18,25 +18,60 @@
 import synthtool.gcp as gcp
 import logging
 import subprocess
+import json
+import os
 
 logging.basicConfig(level=logging.DEBUG)
 
-gapic = gcp.GAPICGenerator()
-version = 'v1'
-library = gapic.node_library(
-        'kms', version,
-        config_path='artman_cloudkms.yaml')
-# skip index, package.json, and README.md
-s.copy(
-    library,
-    excludes=['package.json', 'README.md', 'src/index.js'],
-)
+gapic = gcp.GAPICMicrogenerator()
+versions = ['v1']
+for version in versions:
+    library = gapic.typescript_library(
+            'kms', version,
+            generator_args={
+                "grpc-service-config": f"google/cloud/kms/{version}/cloudkms_grpc_service_config.json",
+                "package-name": f"@google-cloud/kms"
+                },
+            proto_path=f'/google/cloud/kms/{version}',
+            extra_proto_files=['google/cloud/common_resources.proto', 'google/iam/v1']
+            )
+    # skip index, package.json, and README.md
+    s.copy(
+        library,
+        excludes=['package.json', 'README.md', 'src/index.ts', 'src/helper.ts', 'tslint.json'],
+    )
 
 # Copy common templates
 common_templates = gcp.CommonTemplates()
-templates = common_templates.node_library()
+templates = common_templates.node_library(source_location='build/src')
 s.copy(templates)
+# remove unneeded protos in proto_list.json
+list_json='src/v1/key_management_service_proto_list.json'
+remove_proto_keywords=['/google/api', '/google/protobuf', 'google/type']
+with open(list_json, 'r') as f:
```

Comment:
@alexander-fenster @AlanGasperini we had discussed previously ramping down our usage of synth for these kinds of things.  I'm not super comfortable with whatever is happening here (need a comment)

Reply:
I see your pain here. This is one of the unlucky libraries that actually pull another full library (IAM client) and make it a part of its client. We don't support this in micro-generators, and we don't have enough libraries to justify this (non-trivial) work across all languages.

Having said that ^^^, we'll have an inventory of our `synth.py` hacks after the conversion is done, and the libraries that have helpers (three of those at this time: vision, speech, and this one) will have one more look 👀 to see how we can improve their generation. I see some common patterns here.

Reply:
I'm fine with doing it, I just want to make sure it comes with tracked work items and OKRs from the actools side.  This particular set of protos (IAM) have come up across multiple languages, so I think we need a better design here in the long term.  I'm fine with this as a hack as long as we have a long term plan.

Reply:
created an issue here, and we will keep track on that once we have done all the API conversion (30/38 complete now): 
https://github.com/googleapis/gapic-generator-typescript/issues/315